### ISSUE TITLE
Unable to click links in the top half of screen

### ISSUE THREAD

**Issue by mcanivel** [appium/appium#8617]
## The problem

I'm trying to test the web app portion of my hybrid application, and I'm unable to click links in the upper half of the screen. I've ran my test for the web app portion in Safari, and the clicks work fine. I also know that Appium's is in the right context (on the web app vs native), because the click works fine if I manually scroll the link to the lower half of the screen during my test. I have also verified that the WebElement being clicked is the correct anchor element.

## Environment

* Appium version (or git revision) that exhibits the issue: 1.6.4
* Desktop OS/version used to run Appium: macOS Sierra, 10.12.5
* Node.js version (unless using Appium.app|exe): v7.5.0
* Mobile platform/version under test: iOS 10.3
* Real device or emulator/simulator: simulator
* Appium CLI or Appium.app|exe:

## Link to Appium logs

These are the logs for the click: https://gist.github.com/mcanivel/d68bacd40c58f94810ad7b268466c1eb

Unfortunately, I can't provide full screenshots for security reasons, but here are a few that show the anchor's position at the click.

Screenshot of the clickable regions:
![clickregions](https://user-images.githubusercontent.com/6361895/26855694-5e6c69f8-4ad1-11e7-9aa1-f0db2ce80a65.jpg)

Any click in the red area (upper half) doesn't trigger the expected behavior, but if the element is scrolled into the green area (lower half), the app responds to the click.

**Comment by dpgraham** [comment]
Have you confirmed that, as a user, you can click on the upper half and it has an effect?

**Comment by mykola-mokhnach** [comment]
You could calculate screen coordinates of the link control and then perform click by coordinates. Also, it might be possible there is an invisible overlay, which "intercepts" all clicks happening in the upper screen area.

**Comment by mcanivel** [comment]
@dpgraham Yes, I have. I've clicked manually within my test, and that has worked.

@mykola-mokhnach I can try clicking the coordinates. For the overlay, wouldn't an exception be thrown in that case? Something along the lines of: "Element is not clickable at point. Other element would receive the click: ..."

**Comment by mykola-mokhnach** [comment]
@mcanivel it depends on the particular low level driver implementation.

**Comment by jlipps** [comment]
@mcanivel have you tried without `nativeWebTap` set to `true`? It looks like the internal point conversion algorithm is returning an incorrect y-value. This could be because WDA is miscalculating the height of the screen, I'm not sure. But removing that capability should do all the clicks through javascript and avoid any oddities here.
